### PR TITLE
TFKUBE-405: added pre-install script for products

### PR DESCRIPTION
**Pull request by jjeongatl**
## Pull request description

_Provide description for the PR_

## Checklist
- [ ] I have successful end to end tests run (with & without domain)
- [x] I have added unit tests (if applicable)
- [x] I have user documentation (if applicable)

### CODE REVIEW

File: modules/products/jira/locals.tf [hunk 1]
```diff
@@ -49,4 +51,6 @@ locals {
       additionalJvmArgs = ["-Dcom.atlassian.jira.status.index.check=false"]
     }
   }) : yamlencode({})
+
+  license = "AAAB8Q0ODAoPeNp9Uk1z2jAQvetXaKY3ZmSMSRrKjA+JraZkgu2xTTv9Ogh7AVEjeSSZ1P++wpi2CSQHHbQfb9++t+++QIlva4VHY+yOp9eTqTfB9/Mce67nobUCEBtZ16CcR16A0EBLbrgUPo1ymibpLKMoanZLUPFqoUFpf+SiQArDChOxHfjbLUixRkmjig3TEDID/gGbuFdkNEY9at7W0JWH9DN9jBOanjL0d81V27UlI8/9dAKnc8arE3oGag9qFvp39IaSq/CbR97HDzfkfvxhgrZcMadWsmwK4xw+RMuVeWIKHIvD9+Ab1QB6sJmkr7LjWADCvCjJmqUuFK87AbrIBYEurNIxeF2/wWAQxTn5GKckSeNwEeSzOCKLjNqEHyiwu5d42WKzAdyjYCoKWYLCdq8tFAZ/3xhT/5gOh2vpMFMxrTkTTiF3w+rYQeDY8dPBocRCGlxybRRfNgYsMtfYSFw02sidddFBVl5hQDBRnFtgeQUpvc1pSO6+Hkj2Nryh9NmN9ItYzxbil5BPAmU08u0j1657RDq5AOpc7Mww9TfxxtxndbFaM8E16wyYtziQu5qJFnUa29jL4wzhn905aIN7LfFKWuWrZs0FLmEPlbSU9DN+/59NR+8scJnv60dE96xqjtRXrNKA/gDQakzsMCwCFG8JD+9JKcQy6O+SAw69it4/5qBuAhQCz6xRgHMTic87Kqty52VlAGeQxw==X02nj"
```
Comment (nghazali) [inline modules/products/jira/locals.tf:55]: Why do we put license in code? This is a sensitive data and should never expose publicly.

File: modules/products/bitbucket/outputs.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 output "product_domain_name" {
-  value = local.domain_supplied ? "https://${local.product_domain_name}" : "http://${var.ingress.outputs.lb_hostname}/${local.product_name}"
+  value = local.bitbucket_ingress_url
```
Comment (nghazali) [inline modules/products/bitbucket/outputs.tf:2]: 👍

**Review by nghazali — CHANGES_REQUESTED** [review]

File: modules/products/jira/locals.tf (hunk 1 above)
Comment (jjeongatl) [reply modules/products/jira/locals.tf:55]: My bad, removed

File: modules/products/jira/variables.tf [hunk 3]
```diff
@@ -56,8 +56,8 @@ variable "jira_configuration" {
   description = "Jira resource spec and chart version"
   type        = map(any)
   validation {
-    condition = (length(var.jira_configuration) == 6 &&
-    alltrue([for o in keys(var.jira_configuration) : contains(["helm_version", "cpu", "mem", "min_heap", "max_heap", "reserved_code_cache"], o)]))
+    condition = (length(var.jira_configuration) == 7 &&
+    alltrue([for o in keys(var.jira_configuration) : contains(["helm_version", "cpu", "mem", "min_heap", "max_heap", "reserved_code_cache", "license"], o)]))
```
Comment (yzha645) [inline modules/products/jira/variables.tf:60]: Can we make license a standalone variable? We're trying to move away from nested variables, and also license doesn't really fit into "Jira resource spec and chart version" category.

File: modules/products/bitbucket/helm.tf [hunk 4]
```diff
@@ -6,7 +6,7 @@ resource "helm_release" "bitbucket" {
   repository = local.helm_chart_repository
   chart      = local.product_name
   version    = local.bitbucket_helm_chart_version
-  timeout    = 10 * 60 # autoscaler potentially needs to scale up the cluster
+  timeout    = 15 * 60 # autoscaler potentially needs to scale up the cluster
```
Comment (yzha645) [inline modules/products/bitbucket/helm.tf:9]: Does Bitbucket Helm release need to depends_on pre-install job?

File: modules/products/jira/variables.tf (hunk 3 above)
Comment (yzha645) [reply modules/products/jira/variables.tf:60]: Ignore the above comment. I just realised that all other products are doing the same...

File: modules/products/bitbucket/helm.tf (hunk 4 above)
Comment (jjeongatl) [reply modules/products/bitbucket/helm.tf:9]: When I test it, works without dependency.

**Review by yzha645 — APPROVED** [review]

**Review by nghazali — APPROVED** [review]

**Pull request merged**